vm/src/hint_processor/builtin_hint_processor/secp/cairo0_hints.rs

```diff
@@ -0,0 +1,671 @@
+from starkware.cairo.common.cairo_secp.secp_utils import pack
+
+x = pack(ids.x, PRIME) % SECP256R1_P"#;
+pub fn reduce_x(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let x = Uint384::from_var_name("x", vm, ids_data, ap_tracking)?.pack86();
+    exec_scopes.insert_value("x", x.mod_floor(&SECP256R1_P));
+    Ok(())
+}
+
+pub const COMPUTE_Q_MOD_PRIME: &str = r#"from starkware.cairo.common.cairo_secp.secp256r1_utils import SECP256R1_P
+from starkware.cairo.common.cairo_secp.secp_utils import pack
+
+q, r = divmod(pack(ids.val, PRIME), SECP256R1_P)
+assert r == 0, f"verify_zero: Invalid input {ids.val.d0, ids.val.d1, ids.val.d2}."
+ids.q = q % PRIME"#;
+pub fn compute_q_mod_prime(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let val = Uint384::from_var_name("val", vm, ids_data, ap_tracking)?.pack86();
+    let (q, r) = val.div_mod_floor(&SECP256R1_P);
+    if !r.is_zero() {
+        return Err(HintError::SecpVerifyZero(Box::new(val)));
+    }
+    insert_value_from_var_name("q", Felt252::from(&q), vm, ids_data, ap_tracking)?;
+    Ok(())
+}
+
+pub const COMPUTE_IDS_HIGH_LOW: &str = r#"from starkware.cairo.common.math_utils import as_int
+
+# Correctness check.
+value = as_int(ids.value, PRIME) % PRIME
+assert value < ids.UPPER_BOUND, f'{value} is outside of the range [0, 2**165).'
+
+# Calculation for the assertion.
+ids.high, ids.low = divmod(ids.value, ids.SHIFT)"#;
+pub fn compute_ids_high_low(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    exec_scopes.insert_value::<BigInt>("SECP256R1_P", SECP256R1_P.clone());
+
+    const UPPER_BOUND: &str = "starkware.cairo.common.math.assert_250_bit.UPPER_BOUND";
+    const SHIFT: &str = "starkware.cairo.common.math.assert_250_bit.SHIFT";
+
+    let upper_bound = constants
+        .get(UPPER_BOUND)
+        .map_or_else(|| get_constant_from_var_name("UPPER_BOUND", constants), Ok)?;
+    let shift = constants
+        .get(SHIFT)
+        .map_or_else(|| get_constant_from_var_name("SHIFT", constants), Ok)?;
+    let value = Felt252::from(&signed_felt(get_integer_from_var_name(
+        "value",
+        vm,
+        ids_data,
+        ap_tracking,
+    )?));
+    if &value > upper_bound {
+        return Err(HintError::ValueOutside250BitRange(Box::new(value)));
+    }
+
+    let (high, low) = value.div_rem(&shift.try_into().map_err(|_| MathError::DividedByZero)?);
```

Comment:
`div_rem` does not provide the exact behaviour as `divmod`. In cases when either the divisor or the dividend is negative, both methods will differ. I suggest replacing it with `div_mod_floor`.

Reply:
Should be resolved by https://github.com/lambdaclass/cairo-vm/pull/1829#:~:text=fix%3A%20use%20div_mod_floor%20instead%20of%20div_rem

Reply:
I am still curious about `div_mod_floor` and `div_rem`. There are still usages of either even when the hint code is the same. I had touched a little on this with @pefontana. Is there somewhere where this is documented?

Reply:
@FrancoGiachetta Actually the fix is not working as the method is not valid

Reply:
Oh you are right, I thought it was a BigInt. You can dismiss the request then, since a felt is always a a positive value, I guess there's no problem here.

Reply:
I run `make test` and it run, I should have run `make test-extensive_hints`